### PR TITLE
Added an interface for the physical project tree

### PR DESCRIPTION
**Pull request by davkean**
Groups all the common services for the physical project tree. Needed this for the language service host, just front loading it to reduce the amount of code to review.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/roslyn-analysis

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    [Export(typeof(IPhysicalProjectTree))]
+    internal class PhysicalProjectTree : IPhysicalProjectTree
+    {
+        private readonly Lazy<IProjectTreeService> _treeService;
+        private readonly Lazy<IProjectTreeProvider> _treeProvider;
+
+        [ImportingConstructor]
+        public PhysicalProjectTree([Import(ExportContractNames.ProjectTreeProviders.PhysicalProjectTreeService)]Lazy<IProjectTreeService> treeService, [Import(ExportContractNames.ProjectTreeProviders.PhysicalViewTree)]Lazy < IProjectTreeProvider> treeProvider)
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs]: Split to multiple lines.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IProjectTreeServiceFactory.cs [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 {
     internal class IProjectTreeServiceFactory
     {
         public static IProjectTreeService Create(IProjectTree tree)
         {
-            var mock =  new Mock<IProjectTreeService>();
-            
+            var mock = new Mock<IProjectTreeService>();
+
             var treeStateMock = new Mock<IProjectTreeServiceState>();
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IProjectTreeServiceFactory.cs]: IProjectTreeServiceStateFactory.ImplementTree(() => tree)?

**Comment by srivatsn** [comment]
👍

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    [Export(typeof(IPhysicalProjectTree))]
+    internal class PhysicalProjectTree : IPhysicalProjectTree
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs]: I wonder if it's worth using this in the SpecialFileProviders to see if that simplifies the code there.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IProjectTreeServiceFactory.cs (hunk 2 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IProjectTreeServiceFactory.cs]: :+1:

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs (hunk 1 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs]: :+1:

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs (hunk 3 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTree.cs]: I'll attempt this as a follow up.

**Pull request merged**